pandas/core/dtypes/dtypes.py

```diff
@@ -558,24 +554,43 @@ def construct_array_type(cls):
 
     @classmethod
     def construct_from_string(cls, string):
-        """ attempt to construct this type from a string, raise a TypeError if
-        it's not possible
         """
+        Construct a DatetimeTZDtype from a string.
+
+        Parameters
+        ----------
+        string : str
+            The string alias for this DatetimeTZDtype.
+            Should be formatted like ``datetime64[ns, <tz>]``,
+            where ``<tz>`` is the timezone name.
+
+        Examples
+        --------
+        >>> DatetimeTZDtype.construct_from_string('datetime64[ns, UTC]')
+        datetime64[ns, UTC]
+        """
+        msg = "could not construct DatetimeTZDtype"""
         try:
-            return cls(unit=string)
+            match = cls._match.match(string)
+            if match:
+                d = match.groupdict()
+                return cls(unit=d['unit'], tz=d['tz'])
+            else:
+                raise TypeError(msg)
```

Comment:
I think you can remove both of the raise and just pass on the ValueErorr, then raise at the end, e.g.

```
try:
    match = .....
    if match:
      ......return .....
except ValueError:
    pass

raise TypeError(msg)
```

Reply:
maybe first one?